optima/programs.py

```diff
@@ -7,33 +7,13 @@
 """
 
 from optima import OptimaException, printv, uuid, today, getdate, dcp, smoothinterp, findinds, odict, Settings, runmodel, sanitize, objatt, objmeth
-from numpy import ones, max, prod, array, arange, zeros, exp, linspace, append, log, sort, transpose, nan, isnan, ndarray, concatenate as cat
+from numpy import ones, max, prod, array, arange, zeros, exp, linspace, append, log, sort, transpose, nan, isnan, ndarray, concatenate as cat, maximum, minimum
 import abc
-import textwrap
-from pylab import figure, figtext
-from matplotlib.ticker import MaxNLocator
-
-coveragepars=['numtx','numpmtct','numost','numcircum']
-
-
-def vec2budget(progset=None, budgetvec=None):
-    ''' Little helper function to convert a budget vector into a proper budget odict '''
-    
-    # Validate input
-    if None in [progset, budgetvec]: raise OptimaException('vec2budget() requires both a program set and a budget vector as input')
-    if type(progset)!=Programset: raise OptimaException('First input to vec2budget must be a program set')
-    
-    # Get budget structure and populate
-    budget = progset.getdefaultbudget() # Returns an odict with the correct structure
-    if len(budget)==len(budgetvec):
-        for k,key in enumerate(budget.keys()):
-            budget[key] = [budgetvec[k]] # Make this budget value a list so has len()
-    else:
-        errormsg = 'Could not convert budget vector into budget: incompatible lengths (%i vs. %i)' % (len(budgetvec), len(budget))
-        raise OptimaException(errormsg)
-    
-    return budget
 
+# WARNING, this should not be hard-coded!!! Available from
+# [par.coverage for par in P.parsets[0].pars[0].values() if hasattr(par,'coverage')]
+# ...though would be nice to have an easier way!
+coveragepars=['numtx','numpmtct','numost','numcircum'] 
```

Comment:
@cliffckerr agreed. At some stage, it'd be nice it Programs stored actual parameter objects, not just strings that are used to access them. We can do this at a later stage.